lsp/nls/tests/snapshots/main__lsp__nls__tests__inputs__hover_field_typed_block_regression_1574.ncl.snap

```diff
@@ -2,7 +2,9 @@
 source: lsp/nls/tests/main.rs
 expression: output
 ---
-<0:24-0:37>[Applies a function to every element in the given array. That is,
+<0:24-0:37>[```nickel
+(_a -> Number) -> Array _a -> Array Number
```

Comment:
This change  in the LSP is dubious. The user will see an additional monomorphic type instead of the actual polymorphic annotation, which we don't want, I guess. We should try to understand where it comes from. Usually, the typechecker communicates with the LSP through the `TypecheckVisitor` trait, by calling the corresponding `visit_ident` method to update the type affected to a variable or a binding in general. Maybe we call this too late now, once the type has been instantiated? Maybe we call it many times? I don't know, but we should fix that.